File: src/Frontend/projects/core/src/lib/models/dto/calculator/discount-params.ts
export class DiscountParams {
    public id: number;
    public dealId: number;
    public orderType: OrderType;
    public contractType: ContractType;
    public prepayment: number;
    public installment: number;
    public considerMarginality: boolean;
    public commissionContract: string;
    public implementationContract: string;

    static fromJSON(obj: any) {
        if (!obj) {
            return null;
        }

        if (!obj.orderType) {
            obj.orderType = null;
        }

        if (!obj.contractType) {
            obj.contractType = null;
        }

        return Object.assign(
            new DiscountParams(), obj,
        );
    }
}

export enum ContractType {
    Sale = 1,
    Comission = 2,
}

export enum OrderType {
    PreOrder = 1,
    CurrentFreeWarehouse = 2,
    PastFreeWarehouse = 3,
}

File: src/Backend/Common/Models/Deals/DiscountParams.cs
using System;
using Company.Common.Enums;

namespace Company.Common.Models.Deals
{
    public class DiscountParams
    {
        public int Id { get; set; }

        public int DealId { get; set; }

        public virtual Deal Deal { get; set; }

        public DateTime CreateDate { get; set; }

        public int CreatorId { get; set; }

        public OrderType OrderType { get; set; }

        public ContractType ContractType { get; set; }

        public float Prepayment { get; set; }

        public float Installment { get; set; }

        public float HeadDiscount { get; set; }

        public float CeoDiscount { get; set; }

        public bool ConsiderMarginality { get; set; }

        public Guid? ImplementationContract { get; set; }

        public Guid? CommissionContract { get; set; }
    }
}
